w3af/plugins/crawl/url_fuzzer.py

```diff
@@ -269,6 +314,33 @@ def _mutate_file_type(self, url):
                 url_copy.set_extension(filetype)
                 yield url_copy
 
+    def _mutate_file_name(self, url):
+        filename = url.get_file_name()
+        if filename:
+            domain = url.get_domain_path()
+            url_string = domain.url_string
+            name = filename[:filename.rfind(u'.')]
+            extension = url.get_extension()
+
+            mutate_name_testing = (
+                url_string + '#' + filename + '#',
+                url_string + name + ' (copy).' + extension,
+                url_string + name + ' - Copy.' + extension,
+                url_string + name + ' copy.' + extension,
+                url_string + '.~lock.' + filename + '#',
+                url_string + name + '-backup.' + extension,
+                url_string + name + '-bkp.' + extension,
+                url_string + '.' + filename + '.swp',
+                url_string + '_' + filename + '.swp',
+                url_string + '.' + filename + '.swo',
+                url_string + '_' + filename + '.swo',
+                url_string + '~' + filename + '.tmp'
+            )
+
+            for change in mutate_name_testing:
+                newurl = URL(change)
```

Comment:
Just in case, do a try/except around the URL instance creation, we never know what could go wrong when crafting these URLs.

Reply:
I'm not sure if you want the try/except in w3af or if it is just to find out what's generating all `http://localhost/` in the list fuzzable requests.

I did this: 
```python
try :
     newurl = URL(change)
     yield newurl
except ValueError as exception:
     om.out.information('Error while generating a new URL: '\
     + exception)
```

And nothing wrong came from it during all my tests.

Let me know if I have to add the exception to the pull request.

Reply:
Yes, the code you show in the comment is what I wanted, just in case... you never know...

The only change I would propose to that code is to move the `yield newurl` to the `else:` section.